.github/workflows/ci.yaml

```diff
@@ -97,13 +119,45 @@ jobs:
           path: |
             ~/.cargo/
             target/
+            .venv/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
 
       - name: Python Bindings Tests
         run: |
-          just python build
           just python test
 
+  benchmarks:
+    name: Benchmarks (Scale Factor = ${{ matrix.scale_factor }})
+    runs-on: ubuntu-latest-8-cores
+    needs: ["build-python"]
+    # CI on push is only triggered for main branch.
+    if: github.event_name == 'push'
+    strategy:
+      # We want all the benchmark jobs to run one by one.
+      max-parallel: 1
```

Comment:
I would have assumed that "parallel" in this case means "should we run more than one worker at a time" not "should we run more than one job on the same worker"

Reply:
I am assuming we'd have only one worker. Also, IIRC, Github doesn't give us any control over running the next parallel job in a different runner so IG it's for the best.

Reply:
I think we have more workers. 

Why do you think we want only one benchmarks to run one-by-one?

Reply:
My assumption here was we'll move to a self-hosted runner so would want to run it one-by-one. If we have more, we would want to modify this accordingly.